frontend/model/contracts/shared/time.js

```diff
@@ -184,7 +185,8 @@ export function humanDate (
     : ((navigator.languages: any): string[]) ?? navigator.language ?? fallback
   // NOTE: `.toLocaleDateString()` automatically takes local timezone differences into account.
   // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString
-  return new Date(date).toLocaleDateString(locale, options)
+  const dateObj = new Date(date)
+  if (!isNaN(dateObj.valueOf())) return dateObj.toLocaleDateString(locale, options)
```

Comment:
I'm not a fan of this change. Why is this being done?

Is it possible to instead of making the change here, handling any potential issues at the callsite?

Reply:
Because otherwise it'll return 'Invalid Date', which most (all?) of the time isn't what one wants.

Reply:
You didn't answer my question tho...

> Is it possible to instead of making the change here, handling any potential issues at the callsite?

Reply:
> Is it possible to instead of making the change here, handling any potential issues at the callsite?

Not really, because you need to run `new Date()` to know if the value will be invalid or not.